MPChartLib/src/main/java/com/github/mikephil/charting/components/YAxis.java

```diff
@@ -357,12 +378,54 @@ public boolean needsOffset() {
             return false;
     }
 
+    /**
+     * Sets min value restriction for autoscale
+     */
+    public void setAutoScaleMinRestriction(float restrictionValue) {
+        mUseAutoScaleRestrictionMin = true;
+        mAutoScaleMinRestriction = restrictionValue;
+    }
+
+    /**
+     * Sets max value restriction for autoscale
+     */
+    public void setAutoScaleMaxRestriction(float restrictionValue) {
+        mUseAutoScaleRestrictionMax = true;
+        mAutoScaleMaxRestriction = restrictionValue;
+    }
+
+    /**
+     * Resets min value restriction for autoscale
+     */
+    public void resetAutoScaleMinRestriction() {
+        mUseAutoScaleRestrictionMin = false;
+    }
+
+    /**
+     * Resets max value restriction for autoscale
+     */
+    public void resetAutoScaleMaxRestriction() {
+        mUseAutoScaleRestrictionMax = false;
+    }
+
     @Override
     public void calculate(float dataMin, float dataMax) {
 
+        float min = dataMin;
+        float max = dataMax;
+
         // if custom, use value as is, else use data value
-        float min = mCustomAxisMin ? mAxisMinimum : dataMin;
-        float max = mCustomAxisMax ? mAxisMaximum : dataMax;
+        if( mCustomAxisMin ) {
+            min = mAxisMinimum;
+        } else if( mUseAutoScaleRestrictionMin ) {
```

Comment:
Indent this, so that the code checks for `mUseAutoScale???Restriction` ONLY if `mCustomAxisMin` is true, that way it uses the same `mAxisMinum` variable, but it two different possible ways.